crates/iroha_data_model/src/query/builder.rs

```diff
@@ -39,6 +39,7 @@ pub trait QueryExecutor {
     /// # Errors
     ///
     /// Returns an error if the query execution fails.
+    #[expect(clippy::type_complexity)]
```

Comment:
create `QueryResult` (or just `Result`) type instead

Reply:
Can I leave this to you @DCNick3 ? I don't want to introduce new type aliases or structs right now. Also, would you address the newly introduced warnings by #5242 ?